hibernate-core/src/main/java/org/hibernate/internal/log/DeprecationLogger.java

```diff
@@ -249,4 +249,15 @@ void connectionProviderClassDeprecated(
 					"5.3 in upgrading.  It will be removed in a later version."
 	)
 	void logUseOfDeprecatedZeroBasedJdbcStyleParams();
+
+	@LogMessage(level = WARN)
+	@Message(value = "%s has been deprecated",
+			id = 90000025)
+	void deprecatedDialect(String dialect);
+
+	@LogMessage(level = WARN)
+	@Message(value = "%s has been deprecated; use %s instead",
+			id = 90000025)
```

Comment:
I'm surprised this even compiles for you. These ids need to be unique.  The JBoss Logging AP checks that during compile

Reply:
Haha OK sorry I pushed super quickly as I was running out the door.

Reply:
Hrm, strange, for me it compiles and no tests fail.

Perhaps the ids don't need to be unique when you have an overloaded method name like what I have here?

Or perhaps I'm doing something wrong?

Investigating...

Reply:
Seems to work perfectly:

    WARN: HHH90000025: MySQL8Dialect has been deprecated; use org.hibernate.dialect.MySQLDialect instead

Anyway, @sebersole I'm perfectly happy to give them two different `id`s, I gave them the same `id` only because I figured that the represented the same problem (use of a deprecated `Dialect`).

Your call.

Reply:
Maybe it does account for overloaded names, not sure.  I agree it is the same underlying issue and would be nice to reuse the same message id.  Just thought that was not allowed (I know its not in other cases).

@dmlloyd Any thoughts on this?  Thanks

Reply:
Overloading is OK if both messages pertain to the "same" thing.  The definition of "same" is a bit hazy but IMO this case would qualify.

IIRC overloading the same ID is allowed by the tooling as long as the method name matches as well.

Reply:
Awesome to know.  Thanks @dmlloyd !

Reply:
Awesome to know.  Thanks @dmlloyd